source/split-on-rest-element.d.ts

```diff
@@ -0,0 +1,101 @@
+import type {IfNotAnyOrNever, IsExactOptionalPropertyTypesEnabled} from './internal/type.d.ts';
+import type {ApplyDefaultOptions} from './internal/object.d.ts';
+import type {OptionalKeysOf} from './optional-keys-of.d.ts';
+import type {UnknownArray} from './unknown-array.d.ts';
+import type {If} from './if.d.ts';
+
+/**
+{@link SplitOnRestElement} options.
+*/
+type SplitOnRestElementOptions = {
+	/**
+	Whether to preserve the optional modifier (`?`).
+
+	- When set to `true`, the optional modifiers are preserved as-is. For example,
+		`SplitOnRestElement<[number, string?, ...boolean[]], {preserveOptionalModifier: true}>` returns `[[number, string?], boolean[], []]`.
+
+	- When set to `false`, optional elements like `T?` are transformed to `T | undefined` or simply `T` depending on the `exactOptionalPropertyTypes` compiler option. For example:
+		- With `exactOptionalPropertyTypes` enabled,
+			`SplitOnRestElement<[number, string?, ...boolean[]], {preserveOptionalModifier: false}>` returns `[[number, string], boolean[], []]`.
+		- And, with it disabled, the result is `[[number, string | undefined], boolean[], []]`.
+
+	@default true
+	*/
+	preserveOptionalModifier?: boolean;
+};
+
+type DefaultSplitOnRestElementOptions = {
+	preserveOptionalModifier: true;
+};
+
+/**
+Splits an array into three parts,
+where the first contains all elements before the rest element,
+the second is the [`Rest`](https://www.typescriptlang.org/docs/handbook/2/objects.html#tuple-types) element itself,
+and the third contains all elements after the rest element.
+
+Note: If any of the parts are missing, then they will be represented as empty arrays.
+For example, `SplitOnRestElement<[string, number]>` returns `[[string, number], [], []]`,
+where parts corresponding to the rest element and elements after it are empty.
+
+By default, The optional modifier (`?`) is preserved.
+See {@link SplitOnRestElementOptions.preserveOptionalModifier `preserveOptionalModifier`} option to change this behavior.
```

Comment:
Just add the link at the bottom.
```suggestion
@see {@link SplitOnRestElementOptions}
```